zenip-42202.md

```diff
@@ -0,0 +1,121 @@
+
+Please note that this non-ceasing behavior is not enabled for all sidechains v2, but it must be explicitly requested by setting the `withdrawalEpochLength` to “0”.
+
+Sidechains v2 will have two different behaviors based on the creation parameters:
+
+- if the `withdrawalEpochLength` is set to “0”, the sidechain never ceases
+
+- if the `withdrawalEpochLength` is not “0”, the sidechain works exactly as sidechains v1 (with normal ceasing behavior)
+
+## Motivation
+
+The sidechain mechanism that has been included in the Horizen blockchain under the name of “Zendoo“ was designed and implemented having in mind the security of users in a fully decentralized scenario.
+
+There are some use cases where sidechains are implemented in a semi-centralized environment with relaxed security constraints.
+
+In a fully decentralized sidechain, the `Mainchain Backward Transfer Request` (MBTR) guarantees to users the possibility to perform withdrawal of funds directly from mainchain in case the sidechain is somehow compromised (but continues to publish certificates). If the sidechain ceases, users can still retrieve their funds by using the `Ceased Sidechain Withrawal` (CSW) mechanism.
+
+In semi-centralized environments, where users can trust the actors running the sidechain, those withdrawal mechanisms (MBTR and CSW) are not needed and users can rely on the normal Backward Transfer mechanism triggered on the sidechain itself. Moreover, in these environments, having fixed certificate submission windows may lead to unwanted or unexpected ceasing due to a variety of reasons:
+
+- bugs in the software running the nodes
+
+- connectivity issues
+
+- congestion of blocks (too many transactions and certificates for one single block)
+
+**Note**: The changes proposed in this ZenIP can mitigate such risks without affecting the security of the sidechain.
+
+## Specification
+
+The creation of sidechains v2 is triggered by the consensus rules starting with the activation of the fork point #10.
+
+Any attempt to create a sidechain v2 on mainnet before that height will be rejected.
+
+Sidechains v2 will keep all the features included in v1 in addition to the possibility to enable the non-ceasing behavior by properly setting the `withdrawalEpochLength`.
+
+![Non-ceasable sidechain](ZenIP-42202/ZenIP-42202-1.png)
+
+### Constraints
+
+Sidechains v2 will never cease when setting `withdrawalEpochLength` to “0” at creation time. They are not forced to publish certificates during fixed submission windows. However, they can publish a certificate whenever they want, with the only two following constraints:
+
+- The `epoch` values of certificates must be consecutive (i.e. after a certificate of epoch N the sidechain must publish a certificate of epoch N + 1)
+
+- The `quality` mechanism is disabled, so it’s not possible to publish two or more certificates for the same epoch
```

Comment:
If you're referring to the continuity of epochs, it's been done to simplify the implementation for both mainchain and sidechain: on mainchain side, it's easier, for instance, to revert a certificate of epoch E (in case of disconnection of a block) by restoring the certificate of epoch E-1; on sidechain side, allowing to completely skip epochs would require merging the proof for an epoch E with the proof of epoch E+1 and it is currently not possible

If you're referring to the `quality` field, it's been set to 0 to make more evident that the sidechain is working with a different behavior (ceasable sidechains can only provide certificates with quality greater than zero by SDK design). Regardless of the value, it's good and also necessary to disable the "quality" mechanism so that certificates are immediately "final" upon inclusion into a block and the mainchain shouldn't deal with superseding and maturity.

Reply:
Correct, using `version: 2` and `withdrawalEpochLength: 0` when creating a sidechain, we know that it is non-ceasable and in such case we only accept certificates with quality set to 0.